base58/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
+name = "base58"                 # TODO: Needs another name for crates.io
+version = "0.1.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>"]
+license = "CC0-1.0"
+repository = "https://github.com/rust-bitcoin/rust-bitcoin/"
+description = "Bitcoin base58 encoding."
+categories = ["cryptography::cryptocurrencies", "encoding"]
+keywords = ["bitcoin", "base58", "encode", "decode"]
+readme = "README.md"
+edition = "2021"
+rust-version = "1.56.1"
+exclude = ["tests", "contrib"]
+
+[features]
+default = ["std"]
+std = ["hashes/std"]
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
+
+[dependencies]
+hashes = { package = "bitcoin_hashes", version = "0.13.0", default-features = false, features = ["alloc"] }
```

Comment:
Might be interesting to make this optional and disable the check functionality without it. Can be done later.

Reply:
I have never seen a base58 string that was not a base58check string. I think this would increase the test matrix of a crate with minimal value in order to allow something with even less value..

Reply:
OK, that makes sense. However I've also realized it only uses the algorithm, not the newtype and thus depends on `hex` same with keys within address. It's quite weird `Address` depends on hex even though it's not really needed. So maybe we should have `hex` optional (on by default) in `hashes`.

Reply:
Good call, yes, let's make `hex` optional in `hashes`.

Reply:
I had a go at this in #2493